CONTRIBUTING.md

```diff
@@ -30,114 +28,136 @@ Before you get started, please read and follow these guidelines to ensure a smoo
+2. Go to `File > Project Structure > Project` and change to the SDK you downloaded in WSL.
+3. If the SDK does not appear, manually add the SDK.
+4. To find the SDK location, run this command in WSL:
+    **On Ubuntu (WSL):**
+    ```sh
+    which java
+    ```
+IntelliJ IDEA is an integrated development environment (IDE) for Java development. Setting the project SDK ensures that IntelliJ uses the correct Java version for building and running the project.
+
+You can open up WSL in the IntelliJ terminal. Find the down arrow and select ubuntu.
+### Using VS Code (Optional)
+#### Set up WSL Extension in VSCode
+**On Windows:**
+1. Open VSCode extension marketplace, search for and install **WSL**.
+2. On Windows, press `Ctrl+Shift+P` to open the command palette, and run `Shell Command: Install 'code' command in PATH`.
+
+Installing the WSL extension in VSCode allows you to open and edit files in your WSL environment directly from VSCode. Adding the `code` command to your PATH enables you to open VSCode from the WSL terminal.
+#### Verify and Configure Environment Variables
+**On Windows:**
+1. Add VSCode path to the environment variables. The default installation path for VSCode is usually:
+    ```plaintext
+    C:\Users\<Your-Username>\AppData\Local\Programs\Microsoft VS Code\bin
+    ```
+    Replace `<Your-Username>` with your actual Windows username.
+
+    Example:
+
+    ```plaintext
+    C:\Users\epic\AppData\Local\Programs\Microsoft VS Code\bin
+    ```
+Adding VSCode to the environment variables ensures that you can open VSCode from any command prompt or terminal window.
+**On Ubuntu (WSL):**
+```sh
+code --version
+cd gravitino
+code .
+```
+Running `code --version` verifies that the `code` command is available. Using `code .` opens the current directory in VSCode.
+#### Open a WSL Project in Windows VSCode
+**On Ubuntu (WSL):**
+1. **Navigate to Your Project Directory**
+   Use the terminal to navigate to the directory of your project. For example:
+   ```sh
+   cd gravitino
+   ```
+2. **Open the Project in VSCode**
+
+   In the WSL terminal, type the following command to open the current directory in VSCode:
+   ```sh
+   code .
+   ```
+   This command will open the current WSL directory in VSCode on Windows. If you haven't added `code` to your path, follow these steps:
+   
+   - Open VSCode on Windows.
+   - Press `Ctrl+Shift+P` to open the command palette.
+   - Type and select `Shell Command: Install 'code' command in PATH`.
+3. **Ensure Remote - WSL is Active**
+   When VSCode opens, you should see a green bottom-left corner indicating that VSCode is connected to WSL. If it isn't, click on the green area and select `Remote-WSL: New Window` or `Remote-WSL: Reopen Folder in WSL`.
+4. **Edit and Develop Your Project**
+   You can now edit and develop your project files in VSCode as if they were local files. The Remote - WSL extension seamlessly bridges the file system between Windows and WSL.
```

Comment:
I ran into a memory issue when using WSL and I was able to find a solution for it. 

```
### Managing WSL
WSL uses a lot of memory so it is recommended to create a `.wslconfig` file that allows memory limits to be put in place.

1. **Shut down WSL**
If your WSL is open you can shut it down in Windows PowerShell using the following command: `wsl --shutdown`

2. **Navigate to user folder**
Open up file explorer and navigate to `C:\Users\<your-username>`. 

3. **Create the `.wslconfig` file**
Open up notepad or another text editor and input the following:

`[wsl2]
memory=4GB   # Limits VM memory in WSL 2 up to 4GB
processors=4 # Makes the WSL 2 VM use two virtual processors`

#The memory and processor usage can be changed depending on your system's hardware.


4. **Saving the file**
Save the file as `".wslconfig"`. Be sure to include the quotes to let Windows know that this isn't a text file.

```

Reply:
> I ran into a memory issue when using WSL and I was able to find a solution for it. I wrote up the steps I took to resolve my issue.
> 
> ```
> ### Managing WSL
> WSL uses a lot of memory so it is recommended to create a `.wslconfig` file that allows memory limits to be put in place.
> 
> 1. **Shut down WSL**
> If your WSL is open you can shut it down in Windows PowerShell using the following command: `wsl --shutdown`
> 
> 2. **Navigate to user folder**
> Open up file explorer and navigate to `C:\Users\<your-username>`. 
> 
> 3. **Create the `.wslconfig` file**
> Open up notepad or another text editor and input the following:
> 
> `[wsl2]
> memory=4GB   # Limits VM memory in WSL 2 up to 4GB
> processors=4 # Makes the WSL 2 VM use two virtual processors`
> 
> #The memory and processor usage can be changed depending on your system's hardware.
> 
> 
> 4. **Saving the file**
> Save the file as `".wslconfig"`. Be sure to include the quotes to let Windows know that this isn't a text file.
> ```

ok, already updated it into the content